moped-editor/src/utils/mapHelpers.js

```diff
@@ -294,3 +313,54 @@ export function useHoverLayer() {
 
   return { handleLayerHover, featureId, hoveredCoords };
 }
+
+/**
+ * Custom hook that returns a vector tile layer hover event handler and the details to place and populate a tooltip
+ * @param {Object} mapRef - Ref object whose current property exposes the map instance
+ * @param {Object} featureCollection - A GeoJSON feature collection to fit the map bounds around
+ * @return {ViewportStateArray} Array that exposes the setter and getters for map viewport using useState hook
+ */
+/**
+ * @typedef {Array} ViewportStateArray
+ * @property {Object} StateArray[0] - A Mapbox viewport object
+ * @property {Function} StateArray[1] - Setter for viewport state
+ */
+export function useFeatureCollectionToFitBounds(mapRef, featureCollection) {
+  const [viewport, setViewport] = useState(mapConfig.mapInit);
+
+  useEffect(() => {
+    const mapBounds = createZoomBbox(featureCollection);
+    const currentMap = mapRef.current;
+
+    /**
+     * Takes the existing viewport and transforms it to fit the project's features
+     * @param {Object} viewport - Describes the map view
+     * @return {Object} Viewport object with updated attributes based on project's features
+     */
+    const fitViewportToBounds = viewport => {
+      if (featureCollection.features.length === 0) return viewport;
+
+      const featureViewport = new WebMercatorViewport({
+        viewport,
+        width: currentMap._width,
+        height: currentMap._height,
+      });
+      const newViewport = featureViewport.fitBounds(mapBounds, {
+        padding: 100,
+      });
```

Comment:
Coool! This is where the viewport gets its width, size. Zoom is automatically calculated.